### PR TITLE
README: Add documentation on downloads and release cycle

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
During the [VMA 2021.02](https://forum.riot-os.org/t/virtual-maintainer-assembly-2021-02/3111/8) we figured that we need some documentation on how to download RIOT, how our release cycle works, and how fixes for a release are provided. This PR takes care of that by adding a section on that to the README.

### Testing procedure
Read. Check if the markdown is rendered correctly via the Github interface.

### Issues/PRs references
None

### CODE REVIEW

**Comment by miri64** [comment]
Ping?

File: README.md [hunk 1]
```diff
@@ -75,6 +75,47 @@ but not limited to:
 * Sigfox
 * LoRaWAN
 
+## GETTING RIOT
```
Comment (leandrolanzieri) [inline README.md:78]: Wouldn't "Getting RIOT" a part of how to Get Started?

File: README.md [hunk 2]
```diff
@@ -75,6 +75,47 @@ but not limited to:
...
+
+If you prefer things more stable, RIOT provides a quarter annual release cycle.
+The source code for the releases resulting from that cycle you can download [via
+Github][releases] as ZIP file or tarball. You can also checkout a release using
+
+```console
+$ git checkout <YYYY.MM>
+```
+
+in Git, when `<YYYY.MM>` is replaced with the release name.
```
Comment (leandrolanzieri) [inline README.md:97]: ?
```suggestion
in Git, where `<YYYY.MM>` is the release name.
```

File: README.md (hunk 1 above)
Comment (miri64) [reply README.md:78]: "Getting started" in this file is basically just a collection of links to the real documentation, so this did not seem fitting to me. Unless you are asking to add this to the main page of the documentation.

Comment (leandrolanzieri) [reply README.md:78]: I mean the "Getting Started" section. Seems a bit short, and getting the code is part of getting started to me

**Comment by miri64** [comment]
Ping @leandrolanzieri, @kaspar030? IMHO this should go into the release after the VMA this was decided in ;-)

File: README.md (hunk 1 above)
Comment (miri64) [reply README.md:78]: It links to more detailed guides. I have doubts, the links get drowned in the information provided here if it is not its own section.

Comment (leandrolanzieri) [reply README.md:78]: Ok

**Comment by leandrolanzieri** [comment]
Please squash @miri64

**Comment by aabadie** [comment]
Why in the README ? Why not extend the [quickest start section of the documentation](https://doc.riot-os.org/index.html#the-quickest-start) ? The basic git clone is redundant with it btw.

**Comment by miri64** [comment]
> Why in the README ? Why not extend the [quickest start section of the documentation](https://doc.riot-os.org/index.html#the-quickest-start) ? The basic git clone is redundant with it btw.

That was decided in the VMA. In the README it is more prominent and people landing on our GitHub page see it directly.

**Comment by miri64** [comment]
Squashed.

**Comment by leandrolanzieri** [comment]
> The basic git clone is redundant with it btw.

I guess we could remove the clone command line, but I think it is still worth mentioning that a plain clone will get you the newest features and bug fixes.

**Comment by aabadie** [comment]
> That was decided in the VMA

Reading the notes, that's not clear to me. Was it in the 2021.01 release debrief section ? (I just see `Update doxygen/readme first`).
Anyway, I don't think it's worth adding so much details about release cycles in the README. The release description should be part of a dedicated doxygen section/page (maybe linked from the README).

Aren't we supposed to have 2 ACKs when editing the README ?

**Comment by miri64** [comment]
> Reading the notes, that's not clear to me. Was it in the 2021.01 release debrief section ? (I just see `Update doxygen/readme first`).

Yes it is somewhat hidden within.

> - Document Process on the new website / link downloads
> - Leandro: Update doxygen/readme first

IIRC we then decided on README, otherwise, if it would have been up to me, I would have also probably chosen the Doxygen.

> Anyway, I don't think it's worth adding so much details about release cycles in the README. The release description should be part of a dedicated doxygen section/page (maybe linked from the README).

Currently the release cycle is not documented anywhere and it should be transparent as possible to the user. Usually, I think, a user expects bug fixes to come in point releases which is not the case for us necessarily. Other than that, the section just tells the user the options to download RIOT.

**Comment by miri64** [comment]
> Aren't we supposed to have 2 ACKs when editing the README ?

Not sure but seems sensible as it is one of our main entry points for a user or new contributor.

**Comment by aabadie** [comment]
> the section just tells the user the options to download RIOT.

With a lot of details. Too much for a main README to my taste. Keeping the first git clone is OK but we should just point from there to a new documentation page that describes the release cycles.

**Comment by miri64** [comment]
> Keeping the first git clone is OK

Plus download from the release page IMHO.

**Comment by miri64** [comment]
_If_ we go that route. Any more voices?

**Comment by miri64** [comment]
See https://github.com/RIOT-OS/RIOT/pull/16345 for a combined alternative

**Comment by jeandudey** [comment]
#16345 fixed the changes proposed on the README.md and added the doxygen documentation, it would be great if @aabadie or @leandrolanzieri could take a second look before merging that one 😉

**Comment by miri64** [comment]
> #16345 fixed the changes proposed on the README.md and added the doxygen documentation, it would be great if @aabadie or @leandrolanzieri could take a second look before merging that one wink

Mhh only saw that now after merging #16345... Would have been great to have that in there as a warning before merging ;-). 

Anyhow, now that #16345 is merged, this can be closed.

**Pull request closed without merging**